src/filter.rs

```diff
@@ -39,12 +39,29 @@ const SUB_INITIAL: u64 = 0;
 const SUB_ANY: u64 = 1;
 const SUB_FINAL: u64 = 2;
 
-named!(filtexpr<Tag>, alt!(filter | item));
+named!(filtexpr<Tag>, alt!(filter | valuesreturnfilter | item));
```

Comment:
The Search filter __must not__ be extended like this, otherwise one could pass a Matched Values filter to an ordinary Search operation. It would almost certainly be rejected by the server, but that's the wrong place; the user would certainly be confused before tracing the error back to the filter.